File: Basic/Test_ArrangingCoins.py
import sys
INT_MAX = sys.maxsize


def Solution(n):
    if n <= INT_MAX:
        count = 0
        left = n
        for expect in range(1, n+1):
            left = left - expect
            if left >= 0:
                count += 1
            else:
                break
    return count


print(INT_MAX)
print(Solution(5))
print(Solution(10))

File: Basic/Test_ArrangingCoins.c
// https://leetcode.com/problems/arranging-coins/

#include <stdio.h>
#define INT_MAX 2147483647

int Solution(int n)
{
    int count;
    if (n <= INT_MAX)
    {
        count = 0;
        int left = n;
        for (int i = 1; i < n + 1; i++)
        {
            left -= i;
            if (left >= 0)
            {
                count += 1;
            }
            else
            {
                break;
            }
        }
    }
    return count;
}

int main()
{
    int t1 = 5;  // 2
    int t2 = 10; // 4
    printf("%d %d", Solution(t1), Solution(t2));
    return 0;
}